src/app/core/global/global.service.ts

```diff
@@ -24,4 +31,45 @@ export class GlobalService {
             this.statusBar.stopProcess(process, "Failed to fetch platform data.");
         });
     }
+
+    checkForPlatformUpdates() {
+
+        if (!this.checkingForPlatformUpdate) {
```

Comment:
Code style: instead of wrapping the whole function body into a conditional, have a conditional at the top for an early return, and lose the indentation level.
```typescript
if (this.checkingForPlatformUpdate) {
    return;
}
this.checkingForPlatformUpdate = true;
...
```

As for functionality, having an early return like this will make the function do nothing for no apparent reason when called.
In order to prevent this, it can return the check Promise and memoize it, so if another call is made while the existing promise has not completed yet, that same promise is returned. That way, calling the function will consistently yield a result.

https://github.com/rabix/composer/blob/4293c2e443056b5146a1baf87c5394cc1abcf5ec/electron/src/routes.ts#L254
Here's an example of how, if there is an existing call in progress, it adds another callback to its result, but shares the promise itself.